### PR TITLE
find method

### PR DESCRIPTION
**Pull request by AunSyedShah**
returns true if the element is found at the top of the stack

### CODE REVIEW

**Review by ammaratef45 — CHANGES_REQUESTED** [review]
@AunSyedShah According to [this tutorial](https://www.tutorialspoint.com/stack-adt-in-data-structures) the stack isn't supposed to have such functionality.

However, we can implement `contains` to return true if the stack has the object, this behavior is expected in some languages (at least in Java as far as I know)

let me know what you think, also, thanks for being an active contributor 👍

**Comment by AunSyedShah** [comment]
I do not think that there is such method `contains` exist in dart. I know that this is not supposed to have this functionality. Altough, it is a pure suggestion.

**Comment by ammaratef45** [comment]
> I do not think that there is such method `contains` exist in dart. I know that this is not supposed to have this functionality. Altough, it is a pure suggestion.

Yeah, I'm suggesting we implement contains for our stack data structure as an alternative for find

File: lib/stack.dart [hunk 1]
```diff
@@ -33,4 +33,18 @@ class Stack<T> {
...
+  
+  // returns true if element is found at the top of the stack
+  bool find(T x) => (x == _list.last) ? true : false;
+  
+  // returns true if element is found in the stack
+  bool contains(T x) {
+    bool found = false;
+    for (var item in _list) {
+      if (x == item) {
+        found = true;
```
Comment (ammaratef45) [inline lib/stack.dart]: @AunSyedShah you can use `return true;` here and `return false` by the end of the method and no need for the variable `found`

File: lib/stack.dart [hunk 2]
```diff
@@ -33,4 +33,18 @@ class Stack<T> {
 
   /// get the length of the stack.
   int get length => size();
+  
+  // returns true if element is found at the top of the stack
+  bool find(T x) => (x == _list.last) ? true : false;
```
Comment (ammaratef45) [inline lib/stack.dart]: @AunSyedShah can you delete this? I don't think the use of `find` in a stack to be useful as we have the `top` method already

File: lib/stack.dart [hunk 3]
```diff
@@ -33,4 +33,18 @@ class Stack<T> {
...
+  
+  // returns true if element is found in the stack
```
Comment (ammaratef45) [inline lib/stack.dart]: @AunSyedShah 
use `\\\` instead of `\\` for documentation comments

**Review by ammaratef45 — CHANGES_REQUESTED** [review]
@AunSyedShah can you merge the master into your branch (patch-2)?

Here's how:

```
git remote add upstream https://github.com/ammaratef45/dart_stack.git
git fetch upstream
git merge upstream/master
```

you'll find there's a conflict, solve it and add a new commit 👍 

Also, check my comments

File: lib/stack.dart (hunk 2 above)
Comment (AunSyedShah) [reply lib/stack.dart]: deleted

File: lib/stack.dart (hunk 3 above)
Comment (AunSyedShah) [reply lib/stack.dart]: doing it thanks.

File: lib/stack.dart (hunk 1 above)
Comment (AunSyedShah) [reply lib/stack.dart]: done it, thanks friend.

**Comment by ammaratef45** [comment]
@AunSyedShah I'll close this as the #7  is merged already

**Pull request closed without merging**